File: 프로그래머스/2/49993. 스킬트리/스킬트리.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s, vector<string> st) {
    int answer = 0;
    for (int i = 0; i < st.size(); i++) {
        int idx = 0, check = true;
        for (int j = 0; j < st[i].size(); j++)
            if (s.find(st[i][j]) != string::npos)
                if (st[i][j] != s[idx]) {
                    check = false;
                    break;
                } else 
                    idx++;
        if (check)
            answer++;
    }

    return answer;
}

File: 프로그래머스/2/49993. 스킬트리/스킬트리.java
class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;
        
        String regex = "[^" + skill + "]";
        for (String str : skill_trees) {
            String remain = str.replaceAll(regex, "");
            if (skill.startsWith(remain)) {
                answer++;
            }
        }
        
        return answer;
    }
}